src/codeflare_sdk/job/jobs.py

```diff
@@ -108,15 +110,58 @@ def _dry_run(self, cluster: "Cluster"):
             workspace=f"file://{Path.cwd()}",
         )
 
-    def submit(self, cluster: "Cluster") -> "Job":
+    def _missing_spec(spec: str):
+        raise ValueError(f"Job definition missing arg: {spec}")
+
+    def _dry_run_no_cluster(self):
+        return torchx_runner.dryrun(
+            app=ddp(
+                *self.script_args,
+                script=self.script,
+                m=self.m,
+                name=self.name,
```

Comment:
Name is also a required argument here. please add _missing_spec()

Reply:
Ah, is this only the case with `kubenetes_mcad` (is it fine to remain optional with `ray`)?

Reply:
Also odd that it is required when submitting via python, but they don't make you use `--name` via CLI

Reply:
Let me double check, but I'm pretty sure Ray gets the "name" another way to its not a required arg here.

Reply:
> Also odd that it is required when submitting via python, but they don't make you use --name via CLI
Can look into this too.